File: Inverted-Index/src/file_handler/FileReader.java
package file_handler;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class FileReader {

    public List<Doc> getFilesDocs(File[] files) throws FileNotFoundException {
        var docs = new ArrayList<Doc>();
        for (File file : files) {
            docs.add(new Doc(file.getName(), getFilesWords(file)));
        }
        return docs;
    }
    public List<String> getFilesWords(File file) throws FileNotFoundException  {

        try (Scanner scanner = new Scanner(file)) {
            List<String> words = new ArrayList<>();
    
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine().toLowerCase();
                String[] lineWords = line.replaceAll("[^a-zA-Z0-9]"," ").split("\\s+");
                words.addAll(Arrays.asList(lineWords));
            }
            return words;
        }
    }
}

File: Phase5-TDD/InvertedIndex/FileHandler/FileReader.cs
using System.IO;
using System.Collections.Generic;
using System;
using System.Text.RegularExpressions;
using System.Linq;
namespace InvertedIndex.FileHandler
{
    public class FileReader
    {
        public IList<Doc> getFilesDocs(FileInfo[] files)
        {
            var docs = new List<Doc>();
        foreach (var file in files) {
            docs.Add(new Doc(file.Name, getFileWords(file)));
        }
        return docs;
        }

        public IList<string> getFileWords(FileInfo file)
        {

            if(!file.Exists){
                throw new FileNotFoundException();
            }
            var words = new List<string>();
            using (StreamReader reader = file.OpenText()){
                string line = null;
                while((line = reader.ReadLine()) != null){
                    line = line.ToLower();
                    List<string> lineWords = Regex.Replace(line, "[^a-zA-Z0-9]", " ")
                    .Split(' ').Select(word => word.Trim()).Where(word => !string.IsNullOrEmpty(word)).ToList<string>();
                    words.AddRange(lineWords);
                }

            }
            return words;
        }
    }
}